libass/ass_font.c

```diff
@@ -35,6 +42,197 @@
 #include "ass_utils.h"
 #include "ass_shaper.h"
 
+
+/**
+ * Convert a UCS-4 code unit to a packed uint32_t in given multibyte encoding
+ */
+#if defined(_WIN32)
```

Comment:
We usually use `#if defined(_WIN32) && !defined(__CYGWIN__)`.
In this specific case using Microsoft's API might be simpler since we specifically want encodings with Microsoft-specific extensions, but [Cygwin's FAQ](https://cygwin.com/faq/faq.html#faq.programming.win32-api) alludes to some restrictions regarding Win32-API calls so not excluding Cygwin might be bad still.

Reply:
The only used win32 api is `WideCharToMultiByte`, which is in `kernel32.dll`. According to the FAQ, the restrictions are applied to environment variables and filesystem path formats. As far as I know, this function neither get/sets environment variables nor calls any filesystem api (implicitly read from some internal databases maybe, but that should be safe). So I think not excluding Cygwin is OK.

Reply:
Can you add a comment to the code regarding why Cygwin is not included then?